### PR TITLE
Remove ExtLlib dependency

### LINKED ISSUE #32: Remove Extlib dependency
`extlib` conflicts with `base64`. `base64` would probably make for a nicer decoding experience than `Netencoding`, take one more step towards #10. More compelling: extlib advertises itself as deprecated for new code on its repo, and both `mrmime` and `email_message` depend on `base64`, and either would be useful in conjunction with this library.

### PR DESCRIPTION
**Pull request by shonfeder**
Closes #32

Tests are passing and everything builds locally for me, but I haven't yet done any integration testing. I'll pin this branch for a project I'm working on tomorrow and see if it works as expected.

I trust you'll let me know what additional checks I should perform or tests I should add, and if I can improve my approach on any details. :)

### CODE REVIEW

**Comment by astrada** [comment]
Thanks! I'm going to test it against [google-drive-ocamlfuse](https://github.com/astrada/google-drive-ocamlfuse).

File: src/gapi/gapiUtils.ml [hunk 1]
```diff
@@ -77,6 +77,37 @@ let string_before_char c s =
...
+  match String.index_opt s c with
+  | None -> "", s
+  | Some i ->
+    let len = String.length s in
+    let j = i + 1 in
+    String.sub s 0 i, String.sub s j (len - j)
+
+let strip_string s =
+  let leading_trailing_whitespace =
+    Str.regexp "\\(^[ \t\n]+\\|[ \t\n]+$\\)"
```
Comment (astrada) [inline src/gapi/gapiUtils.ml]: I fixed a little bug here, adding `\r` to the regex.

**Comment by astrada** [comment]
So far, it looks good. I'm going to test it a little bit more.

File: src/gapi/gapiUtils.ml (hunk 1 above)
Comment (shonfeder) [reply src/gapi/gapiUtils.ml]: Thanks for catching that :)

**Comment by shonfeder** [comment]
Just reporting that I tested in a little project exercising a (tiny) part of the gmail API, and it's still working as I expected. :)

**Comment by shonfeder** [comment]
Nice! Now a bunch of useful libs can be use along side.

 Thanks for your help here :)

Let me know if you could use any help in cutting a new release to the opam repo.

**Comment by astrada** [comment]
I've already created the PR (ocaml/opam-repository#16632), unfortunately the CI is not working properly: Travis says the generated logs are too long, and the OCaml CI in the failed builds says that the docker image is unavailable.

**Comment by astrada** [comment]
Actually I've closed that and reopened a new one (ocaml/opam-repository#16635) because I fixed another bug.

**Pull request merged**